test.disabled/ejabberd_tests/tests/muc_SUITE.erl

```diff
@@ -2599,16 +2604,98 @@ one2one_chat_to_muc(Config) ->
         escalus_assert:has_no_stanzas(Eve)
     end).
 
+registration_query_service(Config) ->
+    escalus:story(Config, [{alice, 1}, {bob, 1}], fun(_Alice,  Bob) ->
+        escalus:send(Bob, escalus_stanza:to(escalus_stanza:iq_get(<<"jabber:iq:register">>, []), ?MUC_HOST)),
+   	    Resp = escalus:wait_for_stanza(Bob),
+        escalus:assert(is_iq_result, Resp),
+        % assert the response instructs to submit a nick
+        Field = exml_query:path(Resp, [{element, <<"query">>}, {element, <<"x">>}, {element, <<"field">>}]),
+        ?assert_equal(<<"text-single">>, exml_query:path(Field, [{attr, <<"type">>}])),
+        ?assert_equal(<<"nick">>, exml_query:path(Field, [{attr, <<"var">>}])),
+        ok
+    end).
+
+query_register_nick(Nick) ->
+    [
+        #xmlel{name = <<"x">>,
+               attrs = [{<<"xmlns">>, <<"jabber:x:data">>}, {<<"type">>, <<"submit">>}],
+               children = [
+                   #xmlel{name = <<"field">>,
+                          attrs = [{<<"type">>, <<"text-single">>}, {<<"var">>, <<"nick">>}],
+                          children = [
+                              #xmlel{name = <<"value">>,
+                                     children = [#xmlcdata{content = Nick}]}]}
+               ]}
+    ].
+
+
+stanza_register_nick(Nick, To) ->
+    Req = escalus_stanza:iq_set(<<"jabber:iq:register">>,
+                                query_register_nick(Nick)),
+    case To of
+        ?MUC_HOST ->
+            escalus_stanza:to(Req, To);
+        Room ->
+            stanza_to_room(Req, Room);
+        {Room, Nick} ->
+            stanza_to_room(Req, Room, Nick)
+    end.
 
-%%Examples 66-76
-%%Registartion feature is not implemented
-%%TODO: create a differend goruop for the registration test cases (they will fail)
-%registration_request(Config) ->
+try_register_nick(From, Nick, To) ->
+    escalus:send(From, stanza_register_nick(Nick, To)),
+    escalus:wait_for_stanza(From).
+
+assert_registration_response(Resp) ->
+    escalus:assert(is_iq_result, Resp),
+    ?assert_equal(<<"jabber:iq:register">>,
+                  exml_query:path(Resp, [{element, <<"query">>}, {attr, <<"xmlns">>}])).
+
+
+registration_request_service(Config) ->
+    escalus:story(Config, [{alice, 1}, {bob, 1}], fun(Alice,  Bob) ->
+        Resp = try_register_nick(Bob, <<"bylekto">>, ?MUC_HOST),
+        assert_registration_response(Resp),
+        % try again
+        Resp1 = try_register_nick(Bob, <<"bylekto">>, ?MUC_HOST),
+        assert_registration_response(Resp1),
+        % but for Alice it should fail
+        Resp2 = try_register_nick(Alice, <<"bylekto">>, ?MUC_HOST),
+        escalus:assert(is_iq_error, Resp2),
+        ok
+    end).
+
+%%registration_request(Config) ->
```

Comment:
Will we need this?

Reply:
This code is not mine, I only reshuffled it a bit. I think it is here for completeness, but there is no implementation. If you want to discuss it you better contact OD (Original Developer).